odoo/orm/models.py

```diff
@@ -4234,6 +4236,24 @@ def create(self, vals_list: list[ValuesType]) -> Self:
         self = self.browse()
         self.check_access('create')
 
+        # check access to all user-provided fields
+        if not isinstance(vals_list, list):
+            # reify when an iterator is passed as argument
+            vals_list = list(vals_list)
+        vals_fields = {k for vals in vals_list for k in vals}
+        vals_fields.update(
+            field_name
+            for k in self.env.context
+            if k.startswith('default_')
+            and (field_name := k[8:])
+            and field_name in self._fields
+        )
+        for field_name in vals_fields:
+            try:
+                self._check_field_access(self._fields[field_name], 'write')
```

Comment:
Not sure if we really could drop `create` logic here 🤔 
What if I want to create a record, and freeze the field so that no one can modify it after `create` by overriding `_has_field_access`. For example the field will be used as the identifier of the record.

Reply:
It could be a real use case.
Note: existing overwrites only care about read and write.